### PR TITLE
Fix front-end linting errors (4)

### PR DESCRIPTION
**Pull request by axelboc**
Last, big round of front-end linting fixes after removing `incrementalAdoption` from `ui/.eslintrc.js`.

As before, the goal is to enforce reasonably strict linting for new code and to fix obvious issues; it's not to refactor existing code. That's why in most cases I disable rules locally, either at the top of the files or on specific lines.

### CODE REVIEW

File: ui/.eslintrc.js [hunk 1]
```diff
@@ -11,10 +11,11 @@ const dependencies = getDependencies();
 
 module.exports = createConfig({
   cwd: __dirname,
-  incrementalAdoption: true, // turn everything into a warning
   rules: {
-    'unicorn/consistent-destructuring': 'off', // too many failures in React class components
```
Comment (axelboc) [inline ui/.eslintrc.js:16]: now disabled only in React files

File: ui/.eslintrc.js [hunk 2]
```diff
@@ -11,10 +11,11 @@ const dependencies = getDependencies();
...
-    'promise/catch-or-return': 'off', // requires deeper refactoring of fetching layer
```
Comment (axelboc) [inline ui/.eslintrc.js:17]: now disabled locally

File: ui/.eslintrc.js [hunk 3]
```diff
@@ -11,10 +11,11 @@ const dependencies = getDependencies();
...
+    'import/no-default-export': 'off', // too opinionated
+    'import/no-unresolved': 'off', // lots of false positives (perhaps misconfigured?)
+    'simple-import-sort/imports': 'off', // changes global CSS order; must migrate to CSS modules first
```
Comment (axelboc) [inline ui/.eslintrc.js:17]: Another reason why global CSS is bad (especially when imported from each component file): it is hugely dependent on the order in which components are used/imported in the application.

File: ui/.eslintrc.js [hunk 4]
```diff
@@ -24,26 +25,24 @@ module.exports = createConfig({
...
     'sonarjs/elseif-without-else': 'off',
-
-    // The point of `switch` is to be less verbose than if/else-if/else
-    'unicorn/switch-case-braces': ['warn', 'avoid'],
   },
   overrides: [
     createReactOverride({
       ...dependencies,
       rules: {
-        'react/no-unsafe': 'off', // requires non-trivial refactoring
```
Comment (axelboc) [inline ui/.eslintrc.js:41]: now disabled locally

File: ui/src/actions/queue.js [hunk 5]
```diff
@@ -1,9 +1,11 @@
+/* eslint-disable promise/catch-or-return */
+/* eslint-disable promise/prefer-await-to-then */
 /* eslint-disable sonarjs/no-duplicate-string */
 import fetch from 'isomorphic-fetch';
 import { showErrorPanel } from './general';
-import { loadSample } from './sampleChanger';
+import { loadSample } from './sampleChanger'; // eslint-disable-line import/no-cycle
```
Comment (axelboc) [inline ui/src/actions/queue.js:6]: There are a bunch of cyclic imports. Webpack doesn't seem to mind, but it's definitely not ideal...

File: ui/src/components/Equipment/PlateManipulator.jsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/jsx-handler-names */
```
Comment (axelboc) [inline ui/src/components/Equipment/PlateManipulator.jsx:1]: This is quite an important rule, in my opinion, as it enforces The React Way of naming event callback props/functions. So even though there are a _lot_ of failures in the codebase, I decided to disable them manually in every file to try to encourage new code to use the right convention.

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-handler-names.md

#### GOOD

```jsx
<MyComponent onChange={handleChange} />
<MyComponent onChange={props.onSomethingChange} />

<button onClick={handleClick}/>
<button onClick={props.onButtonClick}/>

<form onSubmit={(evt) => {
  evt.preventDefault();
  doSomething();
  props.doSomethingElse();
} />
```

File: ui/src/components/Equipment/SampleChanger.js [hunk 7]
```diff
@@ -95,7 +96,7 @@ export class SampleChangerTreeNode extends React.Component {
     return (
       <div>
         <li className="treeLi">
-          <input
+          <input // eslint-disable-line jsx-a11y/control-has-associated-label
```
Comment (axelboc) [inline ui/src/components/Equipment/SampleChanger.js:99]: Ensuring each control has a label (either in a separate element via `id`+`htmlFor`, or with a simple `aria-label` attribute) makes writing E2E tests a lot simpler (and makes those E2E tests a lot more robust).

File: ui/src/components/SSXChip/SSXChip.jsx [hunk 8]
```diff
@@ -1,3 +1,6 @@
+/* eslint-disable react/jsx-key */
```
Comment (axelboc) [inline ui/src/components/SSXChip/SSXChip.jsx:1]: A lot of warnings related to missing or wrongly provided `key` props. Pretty important to get this right in new code, but too tricky to fix in existing code.

https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key

File: ui/src/components/SSXChip/SSXChip.jsx [hunk 9]
```diff
@@ -70,7 +73,7 @@ export default class SSXChip extends React.Component {
       top_left_z: currentChipLayout.calibration_data.top_left[1],
       top_right_x: currentChipLayout.calibration_data.top_right[0],
       top_right_y: currentChipLayout.calibration_data.top_right[0],
-      top_left_z: currentChipLayout.calibration_data.top_right[1],
+      top_right_z: currentChipLayout.calibration_data.top_right[1],
```
Comment (axelboc) [inline ui/src/components/SSXChip/SSXChip.jsx:76]: bug fix

File: ui/src/components/SSXChip/SSXChip.jsx [hunk 10]
```diff
@@ -1,3 +1,6 @@
+/* eslint-disable react/jsx-key */
+/* eslint-disable react/no-unused-state */
```
Comment (axelboc) [inline ui/src/components/SSXChip/SSXChip.jsx:2]: A bunch of unused states (and prop types). Might be easy to fix but I didn't want to make the call.

File: ui/src/components/SampleQueue/QueueControl.js [hunk 11]
```diff
@@ -133,6 +134,7 @@ export default class QueueControl extends React.Component {
     );
   }
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
```
Comment (axelboc) [inline ui/src/components/SampleQueue/QueueControl.js:137]: Most of the time, this warning is a good clue that components are becoming too big and need to be split into smaller components.

File: ui/src/components/SampleView/shapes.js [hunk 12]
```diff
@@ -305,20 +307,23 @@ export function makePoints(points, imageRatio) {
...
+      }
     }
   }
   return fabricPoints;
 }
 
 export function makeTwoDPoints(points, imageRatio) {
   const fabricPoints = [];
+  // eslint-disable-next-line guard-for-in
   for (const id in points) {
```
Comment (axelboc) [inline ui/src/components/SampleView/shapes.js:322]: Best to use `for ... of`, but the behaviour differs, so didn't want to risk a regression.

File: ui/.eslintrc.js (hunk 3 above)
Comment (marcus-oscarsson) [reply ui/.eslintrc.js:17]: :)

File: ui/src/actions/queue.js (hunk 5 above)
Comment (marcus-oscarsson) [reply ui/src/actions/queue.js:6]: ohh, yes we should fix those

File: ui/src/components/Equipment/PlateManipulator.jsx (hunk 6 above)
Comment (marcus-oscarsson) [reply ui/src/components/Equipment/PlateManipulator.jsx:1]: Thanks, good idea :+1:

File: ui/src/components/Equipment/SampleChanger.js (hunk 7 above)
Comment (marcus-oscarsson) [reply ui/src/components/Equipment/SampleChanger.js:99]: i see

**Comment by marcus-oscarsson** [comment]
Thanks @axelboc, some quite important things to look at.

**Comment by marcus-oscarsson** [comment]
:+1:

**Pull request merged**